src/config/field-overrides/program-indicator/AnalyticsPeriodBoundaries.js

```diff
@@ -51,29 +56,52 @@ const boundaryTypes = [
     },
 ];
 
-// boundary type definitions mapped to java enum
-// dhis2-core/dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/AnalyticsPeriodBoundaryType.java#L37-L40
-// as they are not exposed over the API
 function AnalyticsPeriodBoundary(props) {
     const getTranslation = props.d2.i18n.getTranslation.bind(props.d2.i18n);
+
+    const filteredBoundaryTargets = boundaryTargets
+        .filter(e => e.value !== CUSTOM)
+        .filter(e => e.value === props.boundaryTarget)
+        .length
```

Comment:
The `boundaryTarget` can be:

- one of three enum values
- free-text
- special type of `CUSTOM` which only exists in this context

This creates a filtered list without the `CUSTOM` option and with the overlap between the stored `boundaryTarget` and the enums we have locally. With this we can proceed to figure out what to display.

Reply:
Why not filter in one go? You are iterating through the array twice now. (even thought the array will be shorter after the first filter...
```
.filter(e => e.value !== CUSTOM && e.value === props.boundaryTarget)
```

Reply:
I did it out of readability concerns. I didn't like how the condensed predicate looked.